src/java.base/share/classes/java/util/WeakHashMap.java

```diff
@@ -283,8 +283,14 @@ static Object unmaskNull(Object key) {
      * Checks for equality of non-null reference x and possibly-null y.  By
      * default uses Object.equals.
      */
-    private static boolean eq(Object x, Object y) {
-        return x == y || x.equals(y);
+    private boolean matchesKey(Entry<K,V> e, Object key) {
+        // check if the given entry refers to the given key without
+        // keeping a strong reference to the entry's referent
+        if (e.refersTo(key)) return true;
+
+        // then checks for equality
```

Comment:
Obnoxiously minor nit: plurality is inconsistent. `check if the given entry...` above, and `then check[s] for equality`. Should be `...then check for equality`?

Reply:
OK.  Fixed this grammatical nit.